### DISCUSSION TITLE
Submit a <b-form> when a key is pressed

### DISCUSSION THREAD

**Discussion by lars-seelbach** [cdmoro/bootstrap-vue-3#877] [Q&A] [answered]
Hi,
so I have a <b-form> with a sumbit button. Some input fields are required. I would like to also submit the form when the user presses Ctrl+S. I would like the <b-form> to do the same input validation the submit button triggers.
Is there any way to get this done? I fear not, as the Components do not [defineExpose ](https://vuejs.org/api/sfc-script-setup.html#defineexpose)their functions, right?

```
<script setup lang="ts">

const inputForm = ref(null);

function onKeyDown(e: KeyboardEvent) {
  if ((e.ctrlKey || e.metaKey) && e.key === 's') {
    e.preventDefault();
    onSave(); // <-- This does no validate if the required fields are filled
   inputForm.value?.submit() // <-- This function does not exist. Any other option to trigger the sumbit here?
  }
}
</script>

<template>
  <b-form ref="inputForm" @submit="onSave">...</b-form>
</template>
```

**Comment by VividLemon** [comment]
I'm not sure why that code would not work. The ref should have submit on it automatically. Transferred to an issue

**Comment by VividLemon** [comment, marked as answer]
![image](https://github.com/bootstrap-vue-next/bootstrap-vue-next/assets/31498096/29fdb638-0c30-49da-8c84-a634f060d80d)
